dvc/api.py

```diff
@@ -69,3 +74,75 @@ def _make_repo(repo_url, rev=None):
     else:
         with external_repo(url=repo_url, rev=rev) as repo:
             yield repo
+
+
+def summon(name, fname=None, args=None, repo=None, rev=None):
+    # 1. Read dvcsummon.yaml
+    # 2. Pull dependencies
+    # 3. Get the call and parameters
+    # 4. Invoke the call with the given parameters
+    # 5. Return the result
+
+    with _make_repo(repo, rev=rev) as _repo:
+        fname = fname or "dvcsummon.yaml"
+        summon_path = os.path.join(_repo.root_dir, fname)
+
+        with open(summon_path, "r") as fobj:
+            objects = ruamel.yaml.safe_load(fobj.read()).get("objects")
+            obj = next(x for x in objects if x.get("name") == name)
+            _summon = obj.get("summon")
+
+        outs = [
+            _repo.find_out_by_relpath(dep) for dep in _summon.get("deps", ())
+        ]
+
+        with _repo.state:
+            for out in outs:
+                _repo.cloud.pull(out.get_used_cache())
+                out.checkout()
+
+        with _chdir_and_syspath(_repo.root_dir):
+            call = _import_string(_summon.get("call"))
+
+            if args:
+                _summon.get("args").update(args)
+
+            result = call(**_summon.get("args"))
+
+        return result
+
+
+@contextmanager
+def _chdir_and_syspath(path):
+    # XXX: Some issues with this approach:
+    #   * Not thread safe
+    #   * Import will pollute sys.modules
+    #   * Weird errors if there is a name clash within sys.modules
+    old_dir = os.path.abspath(os.curdir)
+    os.chdir(path)
+    sys.path.insert(0, path)
```

Comment:
You don't need to do both probably. Also, use `try/finally`

Reply:
It is needed, @Suor, or at least I couldn't rely solely on `os.chdir`.
Looks like `importlib.import_module` doesn't care of the current directory (well, `sys.path` is not being updated when running `os.chdir`)

Reply:
By the way, do you mean `try/finally` instead of the context manager, or inside the context manager?